subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -279,6 +199,245 @@ uint8_t ll_sync_create(uint8_t options, uint8_t sid, uint8_t adv_addr_type,
+		return;
+	}
+
+	memcpy(lll->access_addr, si->aa, sizeof(lll->access_addr));
+	lll->data_chan_id = lll_chan_id(lll->access_addr);
+	memcpy(lll->crc_init, si->crc_init, sizeof(lll->crc_init));
+	lll->event_counter = sys_le16_to_cpu(si->evt_cntr);
+
+	interval = sys_le16_to_cpu(si->interval);
+	interval_us = interval * PERIODIC_INT_UNIT_US;
+
+	/* Convert fromm 10ms units to interval units */
+	if (sync->timeout != 0  && interval_us != 0) {
+		sync->timeout_reload = RADIO_SYNC_EVENTS((sync->timeout * 10U *
+						  USEC_PER_MSEC), interval_us);
+	}
+
+	/* Adjust Skip value so that there is minimum of 6 events that can be
+	 * listened to before Sync_Timeout occurs.
+	 * The adjustment of the skip value is controller implementation
+	 * specific and not specified by the Bluetooth Core Specification v5.3.
+	 * The Controller `may` use the Skip value, and the implementation here
+	 * covers a case where Skip value could lead to less events being
+	 * listened to until Sync_Timeout. Listening to more consecutive events
+	 * before Sync_Timeout increases probability of retaining the Periodic
+	 * Synchronization.
+	 */
+	if (sync->timeout_reload > CONN_ESTAB_COUNTDOWN) {
+		uint16_t skip_max = sync->timeout_reload - CONN_ESTAB_COUNTDOWN;
+
+		if (sync->skip > skip_max) {
+			sync->skip = skip_max;
+		}
+	}
+
+	sync->sync_expire = CONN_ESTAB_COUNTDOWN;
+
+	/* Extract the SCA value from the sca_chm field of the sync_info
+	 * structure.
+	 */
+	sca = (si->sca_chm[PDU_SYNC_INFO_SCA_CHM_SCA_BYTE_OFFSET] &
+	       PDU_SYNC_INFO_SCA_CHM_SCA_BIT_MASK) >>
+	      PDU_SYNC_INFO_SCA_CHM_SCA_BIT_POS;
+
+#if defined(CONFIG_BT_CTLR_SYNC_ISO)
+	lll->sca = sca;
+#endif /* CONFIG_BT_CTLR_SYNC_ISO */
+
+	lll->window_widening_periodic_us =
+		DIV_ROUND_UP(((lll_clock_ppm_local_get() +
+				   lll_clock_ppm_get(sca)) *
+				  interval_us), USEC_PER_SEC);
+	lll->window_widening_max_us = (interval_us >> 1) - EVENT_IFS_US;
+	if (PDU_ADV_SYNC_INFO_OFFS_UNITS_GET(si)) {
+		lll->window_size_event_us = OFFS_UNIT_300_US;
+	} else {
+		lll->window_size_event_us = OFFS_UNIT_30_US;
+	}
+
+#if defined(CONFIG_BT_CTLR_DF_SCAN_CTE_RX)
+	lll->node_cte_incomplete = NULL;
+#endif /* CONFIG_BT_CTLR_DF_SCAN_CTE_RX */
+
+	/* Prepare Periodic Advertising Sync Transfer Received event (dispatched later) */
+	sync_handle = ull_sync_handle_get(sync);
+	rx = (void *)sync->node_rx_sync_estab;
+	rx->hdr.type = NODE_RX_TYPE_SYNC_TRANSFER_RECEIVED;
+	rx->hdr.handle = sync_handle;
+	rx->rx_ftr.param = sync;
+
+	/* Create node_rx and assign values */
+	se_past = (void *)rx->pdu;
+	se_past->rx_sync.status = BT_HCI_ERR_SUCCESS;
+	se_past->rx_sync.interval = interval;
+	se_past->rx_sync.phy = sync->lll.phy;
+	se_past->rx_sync.sca = sca;
+	se_past->conn_handle = ll_conn_handle_get(conn);
+	se_past->service_data = service_data;
+
+	conn_interval_us = conn->lll.interval * CONN_INT_UNIT_US;
+
+	/* Calculate offset and schedule sync radio events */
+	ready_delay_us = lll_radio_rx_ready_delay_get(lll->phy, 1);
```

Comment:
```suggestion
	ready_delay_us = lll_radio_rx_ready_delay_get(lll->phy, PHY_FLAGS_S8);
```

Right?

Reply:
Yes, that should be the most correct way. Will change that.

Reply:
yes, please update (atleast in new added code).